Add unit tests for parted module

fridim opened the pull request:
Test the current expected behavior of the module:
- mock parted() and get_device_info()
- Use some of the examples and test the 'script' passed to parted.
- mock check_parted_label() to return false, as if parted version is > 3.1
- assert get_device_info output is correct

Current implementation of the module runs parted several time while going
through all parameters (flags, name, ...). Between calls it uses get_device_info
to update the dictionary. Use check_mode for some of the tests to force module
to go through all the parameters even is dictionary is not updated.

##### SUMMARY
Start the work on unit tests for the parted module. Try not to be too close to the current implementation to allow potential future refactoring.

<!--- Describe the change, including rationale and design decisions -->
In this commit, 2 kind of tests:
- get_device_info  test output is the expected dictionary
- assert script passed to parted is the one expected: use some of the examples. Also one with gpt to test name parameter.
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
- Test Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
modules/system/parted.py
test/units/modules/system/test_parted.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.4.0 (add-parted-unittests eef460febd) last updated 2017/05/01 14:05:05 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/fridim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/fridim/sync/dev/ansible/lib/ansible
  executable location = /home/fridim/sync/dev/ansible/bin/ansible
  python version = 3.6.1 (default, Mar 27 2017, 00:27:06) [GCC 6.3.1 20170306]
```

fridim committed:
Add unit tests for parted module

Test the current expected behavior of the module:
- mock parted() and get_device_info()
- Use some of the examples and test the 'script' passed to parted.
- mock check_parted_label() to return false, as if parted version is > 3.1
- assert get_device_info output is correct

Current implementation of the module runs parted several time while going
through all parameters (flags, name, ...). Between calls it uses get_device_info
to update the dictionary. Use check_mode for some of the tests to force module
to go through all the parameters even is dictionary is not updated.

bcoca commented:
cc @mattclay

mattclay commented:
CI failure in unit tests:

```
>       self.assertEqual(parse_partition_info(parted_output1, 'MB'), parted_dict1)
E       AssertionError: {'gen[241 chars]2', 'name': '', 'flags': ['esp'], 'unit': 'mb'[228 chars]b'}]} != {'gen[241 chars]2', 'flags': ['esp'], 'unit': 'mb'}, {'num': 2[192 chars]b'}]}
E       Diff is 1056 characters long. Set self.maxDiff to None to see it.

test/units/modules/system/test_parted.py:169: AssertionError
```

fridim commented:
@mattclay the test used to pass. Unit key is missing in the dict returned by parse_partition_info() now, since this commit probably https://github.com/ansible/ansible/commit/78fff751aba1ec572a07c3b09169599e8d20def0

Is it the tests that need to be fixed, or is it the code ?

fridim commented:
Cc @ColOfAbRiX

fridim committed:
test_parted.py: add "name" param into expected results

since 78fff751aba1ec572a07c3b09169599e8d20def0, parse_partition_info
fetch the partition name. This commit adds 'name' key and value into
the expected results.

fridim commented:
@mattclay OK, i was fooled by the strange output formating in Shippable. I checked and indeed it's after 78fff75, partition name param was added.  I fixed my unit test to add it.

All checks pass now.
